### PR TITLE
Adding DevContainer with Functions v3, Azure CLI, and Terraform 0.12.24

### LINKED ISSUE #18: [Pre-req] Dev Container
### Description
As Joe, I'd like to have a uniform development environment, so that I know I have all the needed components for building our services.

### Acceptance Criteria
Reference: [Done-Done Checklist] (https://github.com/Microsoft/code-with-engineering-playbook/blob/master/Engineering/BestPractices/DoneDone.md)
- [ ] Should contain Azure Function Core Tools, Azure CLI, Dotnet Core SDK
- [ ] Should run in docker using VSCode devContainer
- [ ] Should be located in starter folder

Also, here are a few points that need to be addressed:

1. n/a

### Resources
[https://code.visualstudio.com/docs/remote/containers](https://code.visualstudio.com/docs/remote/containers)

### Tasks
- [ ] Create Dockerfile with depndancies
- [ ] Create post configuration script
- [ ] Update readme in starter folder

### PR DESCRIPTION
**Pull request by brandonh-msft**
~~~
root@b9801e42f2f5:/workspaces/openhack-durable-entities-coach/starter# dotnet --version
3.1.101
root@b9801e42f2f5:/workspaces/openhack-durable-entities-coach/starter# func --version
3.0.2245
root@b9801e42f2f5:/workspaces/openhack-durable-entities-coach/starter# terraform --version
Terraform v0.12.24
root@b9801e42f2f5:/workspaces/openhack-durable-entities-coach/starter# az -v
azure-cli                          2.5.1

command-modules-nspkg              2.0.3
core                               2.5.1
nspkg                              3.0.4
telemetry                          1.0.4

Python location '/opt/az/bin/python3'
Extensions directory '/root/.azure/cliextensions'

Python (Linux) 3.6.5 (default, Apr 30 2020, 06:22:25) 
[GCC 8.3.0]

Legal docs and information: aka.ms/AzureCliLegal


Your CLI is up-to-date.

Please let us know how we are doing: https://aka.ms/clihats
and let us know if you're interested in trying out our newest features: https://aka.ms/CLIUXstudy
~~~

Fixes #18

### CODE REVIEW

File: starter/.devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        "ms-azuretools.vscode-docker",
+		"ms-azuretools.vscode-azurefunctions",
+        "ms-azure-devops.azure-pipelines",
+        "ms-vscode.azurecli",
+        "ms-vscode.azure-account",
+		"ms-dotnettools.csharp"
+	]
+
+	// Use 'postCreateCommand' to run commands after the container is created.
+	// "postCreateCommand": "dotnet restore",
```
Comment (ianphil) [inline starter/.devcontainer/devcontainer.json:27]: Can we move this out into a script similar to: https://github.com/ianphil/BasicPyDevEnv/blob/a7a4e5566f3a89a184a24a4a6346352b7b1c4e83/.devcontainer/devcontainer.json#L14

Comment (brandonh-msft) [reply starter/.devcontainer/devcontainer.json:27]: oo good idea - hadn't seen that done before 👍🏻👍🏻

File: starter/.devcontainer/devcontainer.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        "ms-azuretools.vscode-docker",
+		"ms-azuretools.vscode-azurefunctions",
+        "ms-azure-devops.azure-pipelines",
+        "ms-vscode.azurecli",
+        "ms-vscode.azure-account",
+		"ms-dotnettools.csharp"
+	],
+
+	// Use 'postCreateCommand' to run commands after the container is created.
+	"postCreateCommand": ".devcontainer/postCommands.sh",
```
Comment (ianphil) [inline starter/.devcontainer/devcontainer.json:27]: should add the `dotnet restore` command to the file :-)

Comment (brandonh-msft) [reply starter/.devcontainer/devcontainer.json:27]: ah yes. 

The command before was commented out, but will add in full dotnet restoring and such

**Pull request merged**